ios-trivia-game/ViewControllers/QuestionViewController.swift

```diff
@@ -11,32 +11,68 @@ import UIKit
+    @IBOutlet weak var questionTitle: UILabel!
+    @IBOutlet weak var answerField: UITextField!
+    @IBOutlet weak var timer: UIBarButtonItem!
+    @IBOutlet weak var submitButton: UIButton!
+    @IBOutlet weak var suggestionLabel: UILabel!
+    
+    fileprivate var countdownTimer = Timer()
+    fileprivate var timerCount = 60
+
     var question: TriviaQuestion?
     
     override func viewDidLoad() {
         super.viewDidLoad()
 
         // Do any additional setup after loading the view.
+        countdownTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(updateCounter), userInfo: nil, repeats: true)
+        timer.title = ""
         
-        // @Nari or @Zhia: Feel free to remove the hardcoded values when you get to this
-        let questionId = 1944
-        FirebaseClient.instance.getQuestionBy(questionId: questionId, complete: { (snapshot) in
-            if let questionArray = snapshot.value as? NSArray {
-                for questionData in questionArray {
-                    if let questionDict = questionData as? NSDictionary {
-                        self.question = TriviaQuestion(dictionary: questionDict)
-                        break
-                    }
-                }
-            }
-            else if let questionDict = snapshot.value as? NSDictionary {
-                self.question = TriviaQuestion(dictionary: questionDict[questionDict.allKeys.first!] as! NSDictionary)
-
+        FirebaseClient.instance.getGameBy(roomId: roomId!, complete: {(snapshot) in
+            let value = snapshot.value
+            let gameRoom = GameRoom(dictionary: value as! NSDictionary)
+            
+            // get the current question
+            let curQuestionIndex = gameRoom.current_question
+            let questions = gameRoom.questions
+            let maxQuestions = gameRoom.max_num_of_questions
+            
+            self.questionNumber.text = "Question \(curQuestionIndex! + 1)/\(maxQuestions)"
+            
+            
+            if (curQuestionIndex! >= questions.count) {
+                // prevent index-out-of-bound error
+                return
             }
-        }, onError: { (error) in
-            Logger.instance.log(logLevel: .error, message: "Could not find question id = \(questionId)")
+            
+            let curQuestionId = questions[curQuestionIndex!]
+            
+            FirebaseClient.instance.getQuestionBy(questionId: curQuestionId, complete: { (snapshot) in
+                if let questionDict = snapshot.value as? NSDictionary {
+                    let question = questionDict.value(forKey: "\(curQuestionId)") as? NSDictionary
+                    self.question = TriviaQuestion(dictionary: question!)
+                }
+                
+                self.questionTitle.text = self.question?.question
+                
+                FirebaseClient.instance.getSuggestionsBy(questionId: curQuestionId, complete: {(snapshot) in
```

Comment:
as discussed, please remove the suggestions code until we complete our basic game